pandas/core/arrays/numpy_.py

```diff
@@ -276,6 +276,15 @@ def __setitem__(self, key, value) -> None:
 
         self._ndarray[key] = value
 
+    def astype(self, dtype, copy: bool = True) -> ArrayLike:
+        if dtype == self.dtype:
+            if not copy:
+                return self
+            elif copy:
+                return self.copy()
+        else:
+            return super().astype(dtype, copy)
```

Comment:
Is it possible to move this to the base class?

Reply:
Yep, tried it and it worked, seems logically consistent. Going to also look to see if I can drop the checks elsewhere by leaving it to the base class.

Reply:
Doesn't seem so, without doing more restructuring than I'm comfortable with ATM.